### PR TITLE
CustomKernelInstall: skip if kernel is already installed

### LINKED ISSUE #453: CustomKernelInstall: skip if kernel is already installed
Testcase CAPTURE-VHD-BEFORE-TEST fails in installation of custom kernel, if the kernel passed as parameter(-CustomKernel) is already installed  in the VM.
This need to be modified to skip if kernel is already installed and CAPTURE-VHD-BEFORE-TEST should be Passed.

07/17/2019 10:48:53 : [ERROR] Kernel upgrade failed in 1 VMs.
07/17/2019 10:48:53 : [ERROR] Custom Kernel: https://lisbuildstorage04.blob.core.windows.net/kernel/3.10.0-957.21.3.el7/kernel-3.10.0-957.21.3.el7.x86_64.rpm installation FAIL. Aborting tests.

build-CustomKernel Logs:
Web link detected. Downloading https://lisbuildstorage04.blob.core.windows.net/kernel/3.10.0-957.21.3.el7/kernel-3.10.0-957.21.3.el7.x86_64.rpm
Custom Kernel:kernel-3.10.0-957.21.3.el7.x86_64.rpm
Removing packages that do not allow the kernel to be installed
Installing kernel-3.10.0-957.21.3.el7.x86_64.rpm
warning: kernel-3.10.0-957.21.3.el7.x86_64.rpm: Header V3 RSA/SHA256 Signature, key ID fd431d51: NOKEY
Preparing...                          ########################################
	package kernel-3.10.0-957.21.3.el7.x86_64 is already installed

### PR DESCRIPTION
**Pull request by DivyaJyothiKasam**
Fixes #453

Added a check to skip kernel installation, if customKernel is already installed and returns true to proceed further installation.

### CODE REVIEW

**Comment by DivyaJyothiKasam** [comment]
Case 1 
Installing the same kernel which is already installed in the VM.
CENTOS 7.6
Initial kernel version: 3.10.0-957.21.3.el7.x86_64
CustomKernel: https://eosgpackages.blob.core.windows.net/testpackages/Kernels/kernel-3.10.0-957.21.3.el7.x86_64.rpm

Logs of build-CustomKernel.txt
yum_install wget: Success
Web link detected. Downloading https://eosgpackages.blob.core.windows.net/testpackages/Kernels/kernel-3.10.0-957.21.3.el7.x86_64.rpm
Custom Kernel:kernel-3.10.0-957.21.3.el7.x86_64.rpm
CUSTOM_KERNEL_ALREADY_INSTALLED

RHEL 7.4
Initial Kernel Version: 3.10.0-693.50.3.el7.x86_64
CustomKernel: https://lisbuildstorage04.blob.core.windows.net/kernel/3.10.0-693.50.3.el7/kernel-3.10.0-693.50.3.el7.x86_64.rpm

Logs of build-CustomKernel.txt
yum_install wget: Success
Web link detected. Downloading https://lisbuildstorage04.blob.core.windows.net/kernel/3.10.0-693.50.3.el7/kernel-3.10.0-693.50.3.el7.x86_64.rpm
Custom Kernel:kernel-3.10.0-693.50.3.el7.x86_64.rpm
CUSTOM_KERNEL_ALREADY_INSTALLED

Case 2:
Upgrade kernel
#1315
Initial Kernel Version: 3.10.0-514.28.1.el7.x86_64
CustomKernel = https://eosgpackages.blob.core.windows.net/testpackages/Kernels/kernel-3.10.0-514.63.1.el7.x86_64.rpm

Logs of build-CustomKernel.txt
yum_install wget: Success
Web link detected. Downloading https://eosgpackages.blob.core.windows.net/testpackages/Kernels/kernel-3.10.0-514.63.1.el7.x86_64.rpm
Custom Kernel:kernel-3.10.0-514.63.1.el7.x86_64.rpm
Removing packages that do not allow the kernel to be installed
Installing kernel-3.10.0-514.63.1.el7.x86_64.rpm
Preparing...                          ########################################
Updating / installing...
kernel-3.10.0-514.63.1.el7            ########################################
Configuring the correct kernel boot order
Updating grub config: /boot/grub2/grub.cfg
CUSTOM_KERNEL_SUCCESS

File: Libraries/CommonFunctions.psm1 [hunk 1]
```diff
@@ -633,6 +633,12 @@ function Install-CustomKernel ($CustomKernel, $allVMData, [switch]$RestartAfterU
 				}
 				return $true
 			} else {
+                $customKernelAlreadyInstall="CUSTOM_KERNEL_ALREADY_INSTALLED"
```
Comment (LiliDeng) [inline Libraries/CommonFunctions.psm1]: indent

File: Libraries/CommonFunctions.psm1 [hunk 2]
```diff
@@ -633,6 +633,12 @@ function Install-CustomKernel ($CustomKernel, $allVMData, [switch]$RestartAfterU
...
+                $currentStatus = Run-LinuxCmd -ip $job.PublicIP -port $job.SSHPort -username $user -password $password -command "tail -n 1 build-CustomKernel.txt"
```
Comment (LiliDeng) [inline Libraries/CommonFunctions.psm1]: this $job is out of the scope line 580 foreach ( $job in $packageInstallJobs ) {

File: Libraries/CommonFunctions.psm1 (hunk 1 above)
Comment (DivyaJyothiKasam) [reply Libraries/CommonFunctions.psm1]: Done with indent changes

**Review by iamshital — APPROVED** [review]

**Review by iamshital — COMMENTED** [review]
Having offline discussion.. Hold the PR..

**Review by iamshital — CHANGES_REQUESTED** [review]
Apply the changes as discussed offline.

**Comment by DivyaJyothiKasam** [comment]
> Apply the changes as discussed offline.

Applied the changes and tested same kernel scenario for  1VM and 2VM setup.
For 2VM setup logs:  pipeline-cloudtest-manual #1705
For 1VM setup logs:  pipeline-cloudtest-manual  #1706
Upgrade kernel scenario: pipeline-cloudtest-manual  #1707
Downgrade kernel : pipeline-cloudtest-manual  #1708

**Review by iamshital — APPROVED** [review]

**Pull request merged**